### ISSUE TITLE
feature request output hostname of azurerm_cdn_endpoint

### ISSUE THREAD

**Issue by dc232** [hashicorp/terraform-provider-azurerm#15144]
### Description
Add ability to export hostname from azure cdn endpoint hostname via output currently there doesn't seem to be a good way to do this current methods suggest (assuming azurerm_cdn_profile is defined)

```
resource "azurerm_cdn_endpoint" "example" {
  name                = var.cdn_endpoint_name
  profile_name        = azurerm_cdn_profile.example.name
  location            = "global"
}
```

then outputting 
```
output "static_website_cdn_endpoint_hostname" {
  value       = azurerm_cdn_endpoint.example.name
  description = "CDN endpoint URL for Static website"
}
```
however, if we were to parse an argument into Cloudflare to create CNAME record to allow us to create a custom domain it would result in 

```
│ Error: failed to create DNS record: HTTP status 400: DNS Validation Error (1004)
```

whilst there does seem to be an attribute called host_name as shown below
```
output "static_website_cdn_endpoint_hostname" {
  value       = azurerm_cdn_endpoint.example.host_name
}
```
it's not very clear as to what hostname it's outputting

The ideal one for Cloudflare would be along these lines

```
$test_var=(az cdn profile list -g test --query [].name -o tsv)
az cdn endpoint list -g test  --profile-name $test_var --query [].hostName -o tsv
```
to produce the needed Cloudflare CNAME to create a custom domain
 
```
az cdn endpoint list -g test  --profile-name $test_var --query [].hostName -o tsv
testingendpont.azureedge.net
````

### New or Affected Resource(s)
affected resource would be
```azurerm_cdn_endpoint```

Can't think of a decent terraform config at the moment and it's not very good practice to use local-exec commands however there probably a better solution, open to suggestion

**Comment by tombuildsstuff** [comment]
hi @dc232 

Thanks for opening this issue.

As you've mentioned the field `host_name` is available which (I'd agree is unclear) does contain the FQDN here (e.g. `{contoso}.azureedge.net`) - so it could be worth renaming this in 3.0 to `fqdn` instead of `host_name` to make this clearer.

Thanks!

**Comment by dc232** [comment]
Ah I see thanks for the speedy reply @tombuildsstuff

**Comment by tombuildsstuff** [comment]
Closing [since per this comment](https://github.com/hashicorp/terraform-provider-azurerm/issues/15144#issuecomment-1105896225) this was released in v3.4.0
